packages/lib/server-only/user/update-password.ts

```diff
@@ -3,50 +3,71 @@ import { compare, hash } from 'bcrypt';
 import { prisma } from '@documenso/prisma';
 
 import { SALT_ROUNDS } from '../../constants/auth';
+import { isPasswordNull } from './is-password-null';
 
 export type UpdatePasswordOptions = {
   userId: number;
   password: string;
-  currentPassword: string;
+  currentPassword?: string;
 };
 
 export const updatePassword = async ({
   userId,
   password,
   currentPassword,
 }: UpdatePasswordOptions) => {
+  console.log('userId', userId, 'update passowrd');
   // Existence check
   const user = await prisma.user.findFirstOrThrow({
     where: {
       id: userId,
     },
   });
 
-  if (!user.password) {
+  if (!user) {
     throw new Error('User has no password');
   }
 
-  const isCurrentPasswordValid = await compare(currentPassword, user.password);
-  if (!isCurrentPasswordValid) {
-    throw new Error('Current password is incorrect.');
-  }
+  const isUserPasswordNull = await isPasswordNull({ id: userId });
```

Comment:
This would mean we're retrieving the user twice, there should be no need for that?